src/components/OptionsList/BaseOptionsList.js

```diff
@@ -77,6 +80,53 @@ class BaseOptionsList extends Component {
         Log.hmmm('[OptionsList] scrollToIndex failed', info);
     }
 
+    /**
+     *
+     * @param {Array} data – this is a flat array of all the sections with all the section headers inline. It comes from react-native in this format
+     * @param {Number} index - the index of the current item in the `data` array
+     * @returns {Object}
+     */
+    getItemLayout(data, index) {
+        const optionHeight = this.props.optionMode === CONST.OPTION_MODE.COMPACT ? variables.optionRowHeightCompact : variables.optionRowHeight;
+        const sectionHeaderHeight = variables.optionsListSectionHeaderHeight;
+
+        let offset = 0;
+        let currentIndex = 0;
+        let isItemSectionHeader = false;
+        _.any(this.props.sections, (section) => {
+            if (currentIndex >= index) {
+                // Stop iteration
+                return true;
+            }
+
+            const startOfNextSection = currentIndex + section.data.length;
+            if (startOfNextSection === index) {
+                isItemSectionHeader = true;
+            }
+
+            const nextIndex = Math.min(startOfNextSection, index);
+
+            if (!section.shouldShow || section.data.length === 0) {
+                // No height is added to the offset by this section, continue to the next section
+                return false;
+            }
+
+            if (section.title && !this.props.hideSectionHeaders) {
+                offset += sectionHeaderHeight;
+            }
+
+            offset += optionHeight * (nextIndex - currentIndex);
+            currentIndex = nextIndex;
+            return false;
+        });
+
+        return {
+            length: isItemSectionHeader ? sectionHeaderHeight : optionHeight,
```

Comment:
**suggestion:** Got a little lost here. Curious if it would just be easier to parse this as a 2D for loop e.g. to find the header indices could we just do something like this?

```js
const headerIndices = [];
for (let i = 0; i < sections.length; i++) {
    for (let j = 0; j < sections[i].data.length; j++) {
        if (j === 0) {
            headerIndices.push(i);
        }
    }
}
```

Reply:
All that said, I agree that the current code with `_.some` is not very readable. Maybe I can improve the readability by just refactoring it to use a normal (non-nested) `for` loop?

Reply:
Ah, I missed that. Does it make a performance difference?

>this function is executed for every item in the list on every render

Even when the items don't change? If so, we could also memoize it, but again - probably would only do this if there is a performance issue.

Mostly just looking for this to be a bit more readable so whatever you think is 0(k) works for me.